<!-- -->
<launch>
  <!-- send table.xml to param server -->
  <arg name="car" default="true" />
  <arg name="pedestrian" default="false" />
  <arg name="obj_car" default="obj_car" />
  <arg name="obj_person" default="obj_person" />


  <group ns="sync_ranging">
    <group if="$(arg car)">
      <group ns="obj_car">
        <node pkg="synchronization" type="sync_range_fusion" name="sync_$(arg obj_car)_ranging">
          <remap from="/image_obj" to="/$(arg obj_car)/image_obj"/>
          <remap from="/vscan_image" to="/vscan_image"/>
          <remap from="/image_obj_ranged" to="/$(arg obj_car)/image_obj_ranged"/>
          <remap from="/sync_ranging/image_obj" to="/sync_ranging/$(arg obj_car)/image_obj"/>
          <remap from="/sync_ranging/vscan_image" to="/sync_ranging/$(arg obj_car)/vscan_image"/>
        </node>
      </group>
    </group>

    <group if="$(arg pedestrian)">
      <group ns="obj_person">
        <node pkg="synchronization" type="sync_range_fusion" name="sync_$(arg obj_person)_ranging">
          <remap from="/image_obj" to="/$(arg obj_person)/image_obj"/>
          <remap from="/vscan_image" to="/vscan_image"/>
          <remap from="/image_obj_ranged" to="/$(arg obj_person)/image_obj_ranged"/>
          <remap from="/sync_ranging/image_obj" to="/sync_ranging/$(arg obj_person)/image_obj"/>
          <remap from="/sync_ranging/vscan_image" to="/sync_ranging/$(arg obj_person)/vscan_image"/>
        </node>
      </group>
    </group>
  </group>
</launch>
